src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -884,13 +911,19 @@ private async Task DoMaintenanceAsync()
 
         var utcNow = SystemClock.UtcNow.AddMilliseconds(50);
 
+        // Remove expired items and items that are infrequently accessed as they may be updated by add.
```

Comment:
For cache entries that are very recently modified, there might be a high churn rate and the cache value might get overridden via Add/Set. So don't touch it in the background clean up job for a little bit. This made the tests much more stable.